pkg/controller/master/setting/controller.go

```diff
@@ -24,3 +47,27 @@ func (h *Handler) LogLevelOnChanged(key string, setting *harvesterv1.Setting) (*
 	logrus.SetLevel(level)
 	return setting, nil
 }
+
+func (h *Handler) overcommitConfigOnChanged(setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
+	overcommit := &settings.Overcommit{}
+	if err := json.Unmarshal([]byte(setting.Value), overcommit); err != nil {
+		msg := fmt.Sprintf("Invalid JSON `%s`: %s", setting.Value, err.Error())
+		return setting, werror.NewInvalidError(msg, "Value")
```

Comment:
use `fmt.Errorf()` probably is more starting forward.

Reply:
Rebased and added `mutator_test.go` for virtualmachine. Thanks for the review!